xiaomi-button/src/xiaomi-utils.lua

```diff
@@ -0,0 +1,68 @@
+-- Copyright 2021 Zach Varberg, SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+local data_types = require "st.zigbee.data_types"
+local capabilities = require "st.capabilities"
+local buf = require "st.buf"
+
+local xiaomi_utils = {}
+
+xiaomi_utils.xiaomi_custom_data_type = {
+  deserialize = function(data_buf)
+    local out = {
+      items = {}
+    }
+    while data_buf:remain() > 0 do
+      local index = data_types.Uint8.deserialize(data_buf)
+      local data_type = data_types.ZigbeeDataType.deserialize(data_buf)
+      local data = data_types.parse_data_type(data_type.value, data_buf)
+      out.items[#out.items + 1] = {
```

Comment:
Why not use `table.insert`?

Reply:
`table.insert` would work fine, this is used in some places as I don't love the implicit assumptions that happen with adding items treating a table as an array and I find this syntax more explicit.  But that was mostly early on before I was as familiar with lua.  I still don't love it, but I've accepted it.  However, because this is duplicated code with the xiaomi-sensors driver, I think I am going to combine them and have the button as a subdriver to avoid that.  I'll close this PR and open another.